### PR TITLE
feat: Add remove_tool_call_params edit strategy

### LINKED ISSUE #73: [Feature]: Add remove_tool_call_params editing strategy
## Describe the solution you'd like
Add a new context editing method to remove the old, long tool-call params with empty json.

## Proposed API/Interface
```python
client.sessions.get_messages(session_id='xxxx', edit_strategies=[
    {"type": "remove_tool_call_params", "params": {...}}
])
```

## Component/Area
Which part of Acontext would this feature affect?
- [x] Client SDK (Python)
- [x] Client SDK (TypeScript)
- [ ] Core Service (Python)
- [x] API Server (Go)
- [ ] UI/Dashboard (Next.js)
- [ ] CLI Tool
- [ ] Documentation
- [ ] Other (please specify)

### PR DESCRIPTION
**Pull request by slyt3**
### Description

Implements #73 - adds remove_tool_call_params edit strategy to reduce context size by removing parameters from old tool calls while keeping recent ones intact.

### Changes

- Go API Server: New strategy in pkg/editor with priority 2
- Python SDK: Added RemoveToolCallParamsStrategy type hints
- TypeScript SDK: Added RemoveToolCallParamsStrategySchema
- Tests: Added unit tests for the new strategy

Tool call ID and name remain intact so tool-results can still reference them. Parameters are replaced with empty JSON '{}'.

### Testing

- Go compilation passes
- Go unit tests pass
- Python syntax validated
- TypeScript compiles successfully
- Tested via API endpoint - verified old tool call parameters are cleared while recent ones are preserved

Example Usage
```
client.sessions.get_messages(
    session_id='xxx',
    edit_strategies=[
        {"type": "remove_tool_call_params", "params": {"keep_recent_n_tool_calls": 3}}
    ]
)
```

### Related Issue

Closes #73

### Tasks

- [x] Implement strategy in Go API
- [x] Add SDK type hints (Python & TypeScript)
- [x] Add unit tests
- [x] Test via API endpoint

### Impact Areas

- [x] Client SDK (Python)
- [x] Client SDK (TypeScript)
- [x] API Server (Go)
- [ ] Core Service
- [ ] Dashboard
- [ ] CLI Tool
- [ ] Documentation
- [ ] Other

### Checklist

- [x] Open pull request against the dev branch
- [x] All tests pass
- [x] Tests added to cover code changes

### CODE REVIEW

File: src/server/api/go/internal/pkg/editor/strategy_remove_tool_call_params.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+	numToModify := totalToolCalls - s.KeepRecentN
+
+	// Remove arguments from the oldest tool-call parts
+	for i := 0; i < numToModify; i++ {
+		pos := toolCallPositions[i]
+		if messages[pos.messageIdx].Parts[pos.partIdx].Meta != nil {
+			messages[pos.messageIdx].Parts[pos.partIdx].Meta["arguments"] = "{}"
+		}
+	}
```
Comment (GenerQAQ) [inline src/server/api/go/internal/pkg/editor/strategy_remove_tool_call_params.go]: - for loop can be modernized using range over int
- (Non-essential) Although the validation layer ensures the existence of Meta, it can handle nil cases for robustness

File: src/server/api/go/internal/pkg/editor/strategy_remove_tool_call_params_test.go
Comment (GenerQAQ) [inline src/server/api/go/internal/pkg/editor/strategy_remove_tool_call_params_test.go:1]: can increase boundary case testing

File: src/server/api/go/internal/pkg/editor/strategy_remove_tool_call_params.go (hunk 1 above)
Comment (slyt3) [reply src/server/api/go/internal/pkg/editor/strategy_remove_tool_call_params.go]: I followed the existing pattern in ```strategy_remove_tool_result.go``` which uses the same `for i := 0; i < numToModify; i++` style for consistency of the codebase. I can modernize if you prefer.

For nil check, I just added it for defensive programming even though the validation layer ensures meta exists, this provides an extra safety  layer that you always validate input/state, if youd prefer to remove it for consistency with other strategies, i can do that too

File: src/server/api/go/internal/pkg/editor/strategy_remove_tool_call_params_test.go
Comment (slyt3) [reply src/server/api/go/internal/pkg/editor/strategy_remove_tool_call_params_test.go:1]: I can add more boundary case tests

- keep_recent_n_tools_calls = 0 (remove all parameters)
- Negative value it should return error
- Messages with no tool calls
- Messages with mix of tool-call and other part types
- TOols call with nil Meta

I will add these boundary cases

File: src/server/api/go/internal/pkg/editor/strategy_remove_tool_call_params.go (hunk 1 above)
Comment (GenerQAQ) [reply src/server/api/go/internal/pkg/editor/strategy_remove_tool_call_params.go]: For the first point, it can be modified; after merging, I will align the other specification with yours.  
The second point can be ignored; the current approach is reasonable for now.
